### ISSUE TITLE
JSS helper components not null when receiving empty layout service field

### ISSUE THREAD

**Issue by jplwood** [Sitecore/jss#162]
## TL;DR
Layout Service/JSS helper components aren't smart enough to be `null` when not in EE or field value is empty.

Potential solutions:
- JSS Helper Components should be Null or a React Component instead of _rendering_ as null when field is null or value is empty (and not in EE)
- Layout service does EE check

## Is your feature request related to a problem? Please describe.
Common issue is that we are passing JSS React components down into vanilla components. 

Say I have a vanilla `Card` component that conditionally renders `body` if it's passed in
```
const Card = ({ body }) => (
  <div>
    <h2>Hello World</h2> {body && <div>{body}</div>}
  </div>
);
```

I might use them like this and in all of these cases, the `div` wrapping `body` will render or not render appropriately:
```
const MyCardSection = () => {
  return (
    <div>
      <Card body={"hi mom"} />
      <Card body={null} />
      <Card />
    </div>
  );
};
```
Now I want to use my `Card` in JSS Sitecore so I create a JSS wrapper component that takes in layout service data and passes in a `RichText` to handle EE. However, when we're working with the layout API and using JSS Components, we lose the ability to know if a value being passed into the card is `null` _because_ we're passing in a JSS Component:
```
const JSSCard = ({ fields }) => {
  const body = <RichText field={fields.body} />;
  return <Card body={body} />;
};
```
Above, `RichText` is we're still passing a valid react component into my `Card`, even if it _will at some point_ render `null`. So when using `Card` in JSS, even if `fields.body` is `{value: ""}`, the `div` tag wrapping my `body` prop inside my card will always render because `RichText` is not `null`, it's a react component that will _render_ as `null`. Note that this is behavior that we _only_ want in EE.

## Describe the solution you'd like
My current thinking is that ideally the EE check would happen in the layout service so we do something like this with confidence:
```
field.body.value == "" ? <RichText field={fields.body} /> : null 
```

Or another potential solution would be to update the JSS React Helper components to be wrapped by a function that either returns `null` or a react component instead of returning a react component that will render as `null` if field is `null` or it's value is `""`.

## Describe alternatives you've considered
A solution to this is that _every_ time I want to use a field from the layout service in a JSS helper component like `Text`, I have to essentially do something like 
```
(fields.body && fields.body.value !== "") || sitecoreContext.pageEditing ? (
  <RichText field={fields.body} />
) : null;

```

That's repetitive and annoying. One little util we came up with to solve this problem is something like this
```
import { isExperienceEditorActive } from "@sitecore-jss/sitecore-jss";

const nullifyFieldsIfNoValueAndNotEditing = fields => {
  return isExperienceEditorActive() ? fields : Object.keys(fields).reduce((newFields, fieldName)=> {
    const field = fields[fieldName];
    const fieldHasValue = field && field.value;
    newFields[fieldName] = fieldHasValue ? field : null
    return newFields
  }, {})
};
```

And just wrap every `fields` in it in our components:
```
const {
  body,
} = nullifyFieldsIfNoValueAndNotEditing(fields);
```

So we can safely just do 
```fields.body ? <RichText field={fields.body} /> : null;```

Obviously still not the best, but a util like this could be useful to include in JSS (or here for reference) if better solutions aren't immediately feasible.

**Comment by sc-illiakovalenko** [comment]
@jplwood You can safely use `getFieldValue` which you can import from `sitecore-jss-react`.
Look at this function [here](https://github.com/Sitecore/jss/blob/dev/packages/sitecore-jss/src/layoutDataUtils.ts#L14)
Example:
```js
import { getFieldValue } from '@sitecore-jss/sitecore-jss-react';

const props = {
  "fields": {
    "heading": {
      "value": ""
    },
    "sample": {
      "value": "111."
    },
    "sample2": {
      "value": "222."
    }
  }
}

console.log('Value: ', getFieldValue(props.fields, 'heading', null))
```

**Comment by sc-illiakovalenko** [comment]
@jplwood Feel free to reopen this issue, if you have same problem
